apps/demo-game/src/reducers/PeriodReducer.ts

```diff
@@ -1,57 +1,55 @@
-import { Action } from '@gbl-uzh/platform'
+import {
+  PayloadPeriodConsolidation,
+  PayloadPeriodInitialisation,
+} from '@gbl-uzh/platform'
 import { debugLog } from '@gbl-uzh/platform/dist/lib/util'
 import { PeriodFacts, PeriodSegmentFacts } from '@graphql/index'
-import { PrismaClient } from '@prisma/client'
-import { match } from 'ts-pattern'
 
-export enum ActionTypes {
-  PERIOD_INITIALIZE = 'PERIOD_INITIALIZE',
-  PERIOD_CONSOLIDATE = 'PERIOD_CONSOLIDATE',
+type State = {
+  rollsPerSegment: number
+  scenario: {
+    seed: number
+    trendStocks: number
+    trendBonds: number
+    gapStocks: number
+    gapBonds: number
+    bankReturn: number
+  }
 }
 
-type Actions =
-  | Action<
-      ActionTypes.PERIOD_INITIALIZE,
-      {
-        periodIx: number
-        periodFacts: PeriodFacts
-        previousPeriodFacts?: PeriodFacts
-        previousSegmentFacts?: PeriodSegmentFacts
-      },
-      PrismaClient
-    >
-  | Action<
-      ActionTypes.PERIOD_CONSOLIDATE,
-      {
-        periodIx: number
-        previousSegmentFacts?: PeriodSegmentFacts
-      },
-      PrismaClient
-    >
+// TODO(JJ):
+// - use immer
+// 1. init baseState outside of fn and provide it as input
+// 2. isDirty is outside of fn
```

Comment:
as discussed, not sure we even need isDirty here, but for consistency we might just keep it (and if we ever allow users to inject stuff into these functions, e.g., with external models, it would be useful)

Reply:
That makes sense. I would suggest to leave it and when we need it we will add it back, though.